spec/controllers/api/v0x0/service_plans_controller_spec.rb

```diff
@@ -1,4 +1,20 @@
 RSpec.describe Api::V0x0::ServicePlansController, :type => :request do
   it("Uses IndexMixin") { expect(described_class.instance_method(:index).owner).to eq(Api::Mixins::IndexMixin) }
   it("Uses ShowMixin")  { expect(described_class.instance_method(:show).owner).to eq(Api::Mixins::ShowMixin) }
+
+  describe "#order" do
+    let(:service_plan) { instance_double("ServicePlan", :id => 123) }
+
+    before do
+      allow(ServicePlan).to receive(:find).with(123).and_return(service_plan)
```

Comment:
I don't think we should stub this

Reply:
I tried to create the `ServicePlan` at first, but it also needs a `Source`, a `Tenant`, and a `ServivceOffering` (which also needs a `Source` and `Tenant`, though I'm guessing they would be the same ones) so I thought I'd rather have the test be fast than create a bunch of objects that (at least for this small spec) don't matter.

I do agree about not stubbing an interface we don't own, but at the same time I think it's ok to not have to worry about the database interaction for this spec. If we had a factory for it I probably would've used that but I think that's probably something reserved for another PR in the future.

Reply:
I'd much rather create the records.

Also, I'm not worried about the amount of time it takes to create the records since the whole test suite currently runs in less than 1 second.

Reply:
But when we're testing with trillions of service plans... yeah lets just create the record :)

Reply:
I know you're exaggerating with the trillions of service plans and yeah it's kind of silly cause we'll never get there, but just for fun after I changed it I did some quick timings. This spec basically goes from about ~0.3ish seconds to ~0.5ish seconds with all of the extra creates. Sure, not too big of a deal in isolation, but even just 100 tests (which I don't think is out of the realm of possibility?) and you're at an extra 20 seconds. Now, we probably don't really care *too* much because most of the time the full suite is just running on travis in the background anyway, but still, it would be really nice to be able to run the full suite locally before committing to a larger codebase and not have to wait a long time.

In the grand scheme of things, 20 seconds or even 200 seconds with 1000 tests isn't really a *huge* deal, I just don't see what actually creating the records is buying us?

I've also had to change the `allow`s a bit to now use an `allow_any_instance_of` instead of stubbing directly on the `service_plan` object because once the `.find` happens in the controller, the spec no longer has a handle to the same object that was created, and it ends up actually trying to call the `#order` method instead of simply returning the canned value.

Reply:
I'd much rather stub the `ServiceCatalogClient` than `ServicePlan.order`.  Then we're testing more than just the route.
```ruby
before do
  allow_any_instance_of(ServicePlan).to receive(:order).with(catalog_parameters).and_return("task_id")
  Rails.application.config.action_dispatch.show_exceptions = false
end
```